tests/testthat/test-tensor.R

```diff
@@ -304,3 +304,15 @@ test_that("element_size works", {
     expect_true(length(result) == 1)
   }
 })
+
+test_that("tensor$bool works", {
+  x <- torch_tensor(c(1,0,1))
+  result <- x$bool()
+  expected <- x$to(torch_bool())
+  expect_equal(result, expected)
```

Comment:
I am not sure what `expect_equal` does, but it doesn't test equality of tensors. We are using `expect_equal_to_tensor(x, y)` to compare tensors in the tests.

Reply:
Changed it. Thanks